default/python/AI/Behavior/Character.py

```diff
@@ -0,0 +1,517 @@
+
+Each AI will get the mandatory behavior (probably Difficulty/Challenge)
+and a selection of the optional behaviors.
+
+The initial implementation was to pull aggression and empire.id modulo 2
+or 3 optional portions of the code into two behaviors: Aggression and
+EmpireID.  These are merely refactoring of existing code and are not
+examples of ideal implementation.
+
+Aggression implements two different behaviors as one: difficulty and
+aggression.  It creates too many decision points not clearly related to
+either concept.  Aggression should be broken into two behaviors:
+difficulty/challenge and aggression.
+
+EmpireID selects based on empire.id with is an implementation detail and
+not a "visible" game mechanic.  EmpireID is also used to select between
+coefficient options, which look like programmer experiments.  EmpireID
+should be removed as a behavior.
+
+Some ideas for future behavior modules are:
+challenge/difficulty    -- to replace the difficulty related portions of
+                           aggression.
+
+research bent -- models an obsession/focus/repulsion by a particular
+area of research. i.e. Shields are for grues.
+
+nemesis/single-minded -- model a focus on a single opponent, perhaps
+the first one to attack the AI.  This plays well with CharacterStrings
+which can be used to inform all players who care that "I will hunt the
+Dominion to the ends of the galaxy."  Humans and eventually AIs could
+use this to modify their risk assessment of this AI.
+
+deceitful/trustworthy -- Only useful after treaties are implemented to
+determine if this AI will abide by the treaty
+
+friendly and loyal -- Even without treaties this AI will assist their
+friends.
+
+vengeful -- An eye for eye.  AI generally counter attacks until things
+are made right or even.
+
+risk averse/gambler -- How much overkill does the AI require?  How risky
+a research strategy will it attempt?  How many redundant buildings does
+the AI build?
+
+warlike/peaceful -- Will the AI prefer colonization to war?
+
+cooperative/loner -- Once cooperative treaties (joint attack, sharing
+system) are implemented, will the AI use them?
+
+taciturn/chatty -- How often does the AI chat?
+
+genocidal -- Will the AI research and use Concentration Camps/bombard
+weapons/planet and star destroying weapons?  Is the AI horrified by the
+use of such weapons?  Will the AI band together with other like minded
+empires to protect the galaxy?
+
+per playable species -- There is no need for per species behavior.  Each
+playable species can be assigned some additional mandatory behavior(s).
+For example the Trith and a mandatory Genocidal behavior.  Perhaps add
+a probabilty distribution of behavior components to the FOCS description
+of a playable species.
+"""
+
+import abc
+from collections import defaultdict, Counter
+import random
+
+import freeOrionAIInterface as fo  # pylint: disable=import-error
+
+
+class Behavior(object):
```

Comment:
Do you mean to replace `Behavior` everywhere with `Trait`?

I'm OK with that, but I want to double check before I change it everywhere.